subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -520,11 +525,18 @@ You can then register it for incremental compilation using the `getSupportedOpti
 </sample>
 ++++
 
-Processors that don't fit these categories will result in full recompilation.
-This includes processors that use `java.io` instead of the `Filer` API and processors that need to read or write resource files.
+Both categories have the following limitations:
+
+* Must not have any side effects except for using the `Filer` and `Messager` APIs.
+For instance, writing files using `java.io` will result in silent failures later on, as these files won't be cleaned up correctly.
+If your processor does this, it cannot be incremental.
+* Must not depend on compiler-specific APIs like `com.sun.source.util.Trees`. Gradle wraps the processing APIs, so attempts to cast to compiler-specific types will fail.
+If your processor does this, it cannot be incremental, unless you have some fallback mechanism.
+* Using `Filer#getResource` currently results in full recompilation. See https://github.com/gradle/gradle/issues/4701[gradle/issues/4701]
+* Using `Filer#createResource` currently results in full recompilation. See https://github.com/gradle/gradle/issues/4702[gradle/issues/4702]
```

Comment:
How is the preceding line tied to this list? Does the "does this" refer to the items in the list?

Reply:
This belongs to the previous point. I think this is my markdown habits coming through, because in Markdown things don't work if you use indentation. Seems to work in asciidoc though, so I'm gonna change it.